### PR TITLE
EVG-1288 Task execution data files only appear when viewing Latest

### PR DESCRIPTION
**Pull request by richardsamuels**

### CODE REVIEW

File: plugin/builtin/attach/attach_plugin.go [hunk 1]
```diff
@@ -74,12 +74,24 @@ func (self *AttachPlugin) GetPanelConfig() (*plugin.PanelConfig, error) {
 					if context.Task == nil {
 						return nil, nil
 					}
-					artifactEntry, err := artifact.FindOne(artifact.ByTaskId(context.Task.Id))
+					taskId := context.Task.Id
+					if context.Task.OldTaskId != "" {
+						taskId = context.Task.OldTaskId
+					}
+
+					artifactEntry, err := artifact.FindOne(artifact.ByTaskIdAndExecution(taskId, context.Task.Execution))
```
Comment (bsamek) [inline plugin/builtin/attach/attach_plugin.go]: Can you add a test for with with/without execution find logic?

File: model/artifact/artifact_file_test.go [hunk 2]
```diff
@@ -5,82 +5,81 @@ import (
...
+	s.NoError(db.Clear(Collection))
+	s.testEntry = Entry{
+		TaskId:          "task1",
+		TaskDisplayName: "Task One",
+		BuildId:         "build1",
+		Files: []File{
+			{"cat_pix", "http://placekitten.com/800/600", ""},
+			{"fast_download", "https://fastdl.mongodb.org", ""},
+		},
+		Execution: 1,
```
Comment (bsamek) [inline model/artifact/artifact_file_test.go]: What sets an `Entry`'s execution number in the first place?

File: plugin/builtin/attach/attach_plugin.go (hunk 1 above)
Comment (richardsamuels) [reply plugin/builtin/attach/attach_plugin.go]: Search the diff for "$setOnInsert". The first upsert (i.e. insert) will set the execution number.

File: model/artifact/artifact_file_test.go (hunk 2 above)
Comment (richardsamuels) [reply model/artifact/artifact_file_test.go]: Search the diff for "$setOnInsert". The first upsert (i.e. insert) will set the execution number.

Comment (bsamek) [reply model/artifact/artifact_file_test.go]: I meant, what sets the execution number in the struct? (not in the db)

Comment (richardsamuels) [reply model/artifact/artifact_file_test.go]: @bsamek https://github.com/evergreen-ci/evergreen/pull/406/files#diff-4b09415ebcdfdcbfcd1446bf22e978c2R364

Comment (bsamek) [reply model/artifact/artifact_file_test.go]: thanks

File: plugin/builtin/attach/attach_plugin.go (hunk 1 above)
Comment (richardsamuels) [reply plugin/builtin/attach/attach_plugin.go]: done.

**Pull request merged**